### PR TITLE
Shared content stat

### PR DESCRIPTION
**Pull request by anpolimus**
Steps for testing:
1. - [ ] Enable shared content server module at one of the builds. Make all content available to share at `admin/content` page
2. - [ ] Add the first server as a source server instead of shared.openy at: `admin/virtual-y/shared-content/server/add`
3. - [ ] Open fetch form at the second server and try to download content to your local Virtual Y
4. - [ ] Verify that count of downloads increased after successful download at the first server reporting page: `/admin/virtyal-y/shared-content/download-stat`
5. - [ ] Verify that count of downloads updated at the fetch form.
6. - [ ] Check filters and the ability to export in csv at the first server reporting page.

### CODE REVIEW

**Review by hamrant — CHANGES_REQUESTED** [review]

File: modules/openy_gc_shared_content/src/Plugin/SharedContentSourceType/VirtualYBlogPost.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function getTeaserJsonApiQueryArgs() {
    */
   public function getFullJsonApiQueryArgs() {
     return [
-      'include' => 'field_vy_blog_image,field_vy_blog_image.field_media_image',
+      'include' => 'field_vy_blog_image,field_vy_blog_image.field_media_image, field_share_count, field_gc_origin',
```
[the remark this answers is not in the document]
Comment (anpolimus) [reply modules/openy_gc_shared_content/src/Plugin/SharedContentSourceType/VirtualYBlogPost.php:36]: Ah yes, you are right

File: modules/openy_gc_shared_content_server/openy_gc_shared_content_server.routing.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+openy_gc_shared_content_server.increase_downloads_stat:
+  path: '/virtual-y-server/inc-downloads'
+  defaults:
+    _controller: '\Drupal\openy_gc_shared_content_server\Controller\DownloadsIncrementController::execute'
+  methods: [POST]
+  requirements:
+    _permission: 'access content'
```
[the remark this answers is not in the document]
Comment (anpolimus) [reply modules/openy_gc_shared_content_server/openy_gc_shared_content_server.routing.yml:7]: Token now checks inside of the post

[the remark this answers is not in the document]
Comment (hamrant) [reply modules/openy_gc_shared_content_server/openy_gc_shared_content_server.routing.yml:7]: ok, I see

File: modules/openy_gc_shared_content/src/SharedContentSourceTypeBase.php [hunk 3]
```diff
@@ -194,6 +196,34 @@ public function jsonApiCall($url, array $query_args = [], $uuid = NULL) {
...
+        ->getQuery()
+        ->condition('url', $url)
+        ->execute());
+
+      $source = SharedContentSource::load($server_id);
+
+      $downloads_stat_push = $this->client->post($url . '/virtual-y-server/inc-downloads', [
+        'form_params' => [
+          'uuid' => $uuid,
+          'url' => 'http://openy.docksal',
```
[the remark this answers is not in the document]
Comment (anpolimus) [reply modules/openy_gc_shared_content/src/SharedContentSourceTypeBase.php]: already :)

File: modules/openy_gc_shared_content_server/config/install/views.view.virtual_y_shared_content_statistic.yml [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+  default:
+    display_plugin: default
+    id: default
+    display_title: Master
+    position: 0
+    display_options:
+      access:
+        type: perm
+        options:
+          perm: 'access content'
```
[the remark this answers is not in the document]
Comment (anpolimus) [reply modules/openy_gc_shared_content_server/config/install/views.view.virtual_y_shared_content_statistic.yml]: updated

**Review by hamrant — CHANGES_REQUESTED** [review]
```
Unable to install Open Y Virtual Y shared content server due to unmet dependencies: views.view.virtual_y_shared_content_statistic (csv_serialization, views_data_export)
```

Error on module enabling (looks like we need this dependencies on module level)

**Comment by anpolimus** [comment]
retest

**Review by hamrant — APPROVED** [review]

**Pull request merged**